### PR TITLE
Added new version warning for repositories in rit helper

### PR DESCRIPTION
**Pull request by brunats**
**- What I did**

Added new version notice available for a repository.

> rit

**- How to verify it**

> Clone this pr with the command
```bash
git fetch upstream pull/582/head:improvement/added-new-version-notice-in-helper
git checkout improvement/added-new-version-notice-in-helper
make build && cp dist/linux/rit /usr/local/bin/
```
> Add a test repository with an outdated version (eg https://github.com/brunats/brunats-formulas and version v1.0.3)
```bash
rit add repo
rit list repo # Check the repo added
```

> Check the new version notice in the repository name prefix
```bash
rit
```

**- Description for the changelog**

Added new version warning for repositories in rit helper

### CODE REVIEW

File: pkg/formula/tree/default_tree.go [hunk 1]
```diff
@@ -99,11 +101,21 @@ func (d Manager) MergedTree(core bool) formula.Tree {
 		if err != nil {
 			continue
 		}
+		noticeNewVersion := ""
+		if d.isRootCommand {
```
Comment (brunasilvazup) [inline pkg/formula/tree/default_tree.go:105]: With this condition, only in the root command will requests be made to check for new versions of the repo.

**Review by lucasdittrichzup — APPROVED** [review]

**Review by GuillaumeFalourd — APPROVED** [review]

**Comment by GuillaumeFalourd** [comment]
Loved the contribution @brunats .

A possible enhancement could be to color the message, as it may not be detected by the user the way it is now.
Something in Red or Yellow as the rit version warning for example.

I'm also wondering, what would be the complexity to implement something like this:
```
A new version of DEMO repository is available, would you like to update it ? (Yes/No/Ignore)
```
And if the answer is Yes, the CLI could execute the `rit update repo` command with the latest version automatically.

Does it make sense ? Is it worth it ?

**Comment by brunasilvazup** [comment]
> Loved the contribution @brunats 
> 
> A possible enhancement could be to color the message, as it may not always be detected by the user the way it is now.
> Something in Red or Yellow as the rit version warning for example.
> 
> I'm also wondering, what would be the complexity to implement something like this:
> 
> ```
> A new version of DEMO repository is available, would you like to update it ? (Yes/No)
> ```
> 
> And if the answer is Yes, the CLI could execute the `rit update repo` command with the latest version automatically.
> And if the answer is No, the CLI wouldn't ask to update the repo for this version anymore.
> 
> Does it make sense ? Is it worth it ?

Hi @GuillaumeFalourd,
About the change of text color may be possible, i will be check the possibility of implementation.

About the other solution, I already have doubts about the cost (_both in terms of implementation and user experience_) and benefit (_how much would be the extra gain and if it is really necessary_).
I believe it would be good to measure the impact of the current solution and see if it is already effective enough.

**Comment by GuillaumeFalourd** [comment]
No problem, I approved the PR as I think it answers the related ISSUE 🚀 
Check the color part then, and leave the rest for later, it's not a priority for now 😄 👍

**Review by lucasdittrichzup — APPROVED** [review]

**Review by GuillaumeFalourd — APPROVED** [review]

**Comment by brunasilvazup** [comment]
\merge qa

**Pull request merged**